compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ProjectDirs.java

```diff
@@ -61,4 +65,22 @@ public static BLangCompilerException getPackageNotFoundError(String sourcePackag
 
         return new BLangCompilerException("cannot find package '" + sourcePackage + "'");
     }
+
+    /**
+     * Checks if the package contains ballerina source files.
+     *
+     * @param pkgPath package path
+     * @return true if source files exists else false
+     */
+    public static boolean containsSourceFiles(Path pkgPath) {
+        List<Path> sourceFiles = new ArrayList<>();
+        try {
+            sourceFiles = Files.find(pkgPath, Integer.MAX_VALUE, (path, attrs) ->
```

Comment:
We are not walking into the subfolders ?

Reply:
Files.find will walk into the sub-folders as https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#find-java.nio.file.Path-int-java.util.function.BiPredicate-java.nio.file.FileVisitOption...-